Library/Homebrew/exceptions.rb

```diff
@@ -65,6 +65,25 @@ def to_s; <<-EOS.undent
   end
 end
 
+class TapFormulaAmbiguityError < RuntimeError
+  attr_reader :name, :paths, :formulae
+
+  def initialize name, paths
+    @name = name
+    @paths = paths
+    @formulae = paths.map do |path|
+      path.to_s =~ HOMEBREW_TAP_PATH_REGEX
+      "#{$1}/#{$2.sub("homebrew-", "")}/#{path.basename(".rb")}"
+    end
+
+    super <<-EOS.undent
+      Multiple formulae found: #{formulae.map { |f| "\n       * #{f}" }.join}
```

Comment:
Perhaps just push the prioritisation to another PR as this one better matches current behaviour. I think we should require the fully-qualified name. This will require formulae changes in taps so should be introduced with a warning for quite a while before we start erroring out.